Jenkinsfile

```diff
@@ -4,30 +4,75 @@ pipeline {
         CGO_ENABLED = 0
         GO111MODULE = 'on'
         SLACK_MESSAGE = "Protos CI » <${RUN_DISPLAY_URL}|Jenkins ${BRANCH_NAME} Job>${ env.CHANGE_URL ? " » <${CHANGE_URL}|GitHub PR #${CHANGE_ID}>" : '' }"
+        GOPATH = "${env.WORKSPACE}/GOPATH"
+        GOBIN = "${env.GOPATH}/bin"
+        PROTOC_HOME = "${env.WORKSPACE}/PROTOC_HOME"
+        PATH = "${env.PROTOC_HOME}/bin:${env.GOBIN}:${env.PATH}"
+        PROTOC_VERSION = "3.19.4"
     }
 
     stages {
-	stage('unit tests') {
+        stage('Config') {
+            steps {
+                sh 'printenv'
+                echo "params=${params}"
+            }
+        }
+        stage('unit tests') {
             options { retry(3) }
             steps {
                 sh 'go test -v ./... 2>&1 | tee unit-test-results.txt && cat unit-test-results.txt | go-junit-report > vega-unit-test-report.xml'
                 junit checksName: 'Unit Tests', testResults: 'vega-unit-test-report.xml'
             }
-	}
+        }
+        stage('Install dependencies') {
+            options { retry(3) }
+            steps {
+                sh label: 'protoc', script: """#!/bin/bash -e
+                    PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+                    curl -LO \$PB_REL/download/v${env.PROTOC_VERSION}/protoc-${env.PROTOC_VERSION}-linux-x86_64.zip
```

Comment:
The other tools we use (buf, protoc-gen-go etc) get installed by ansible here:

https://github.com/vegaprotocol/ansible/blob/a2f36ea1194abe15bb3ac221e4d94807c16877c0/roles/jenkins-agent/tasks/main.yaml#L191

I don't really mind where we do it, but it would be nice I think to keep them all in the same place. Random thought - maybe the optimal solution would be for the CI would to call `gettools.sh` then everything would in in one place; though removing stuff from ansible would affect core & datanode builds too so that's probably more faff than its worth.

Reply:
Yes, @fkondej and me explored different solutions on where to install these deps. Do we call `buf` and `protoc` in the core and the data-node ? I thought we only called them in this repo. That's the reason we decided to reference them here. Are you sure these are needed by core and data-node ?

Reply:
Resolving just to unblock the merge request. Re-opening after.

Reply:
We only use them in `protos` (though I think that wasn't true in the past), so I think it's fine to have them in the Jenkins file here - my minor complaint is that now we've got protoc being installed here in the Jenkins pipeline, but buf and protoc-gen-go in the ansible playbook which is a little confusing.